packages/Parallel/WorkerRunner.php

```diff
@@ -93,11 +94,12 @@ public function run(Encoder $encoder, Decoder $decoder, Configuration $configura
                     $file = new File($filePath, FileSystem::read($filePath));
                     $this->currentFileProvider->setFile($file);
 
-                    $errorAndFileDiffs = $this->processFile($file, $configuration, $errorAndFileDiffs);
+                    $processFileResult = $this->processFile($file, $configuration, $errorAndFileDiffs);
+                    $errorAndFileDiffs = $processFileResult->getErrorAndFileDiffs();
 
                     if ($errorAndFileDiffs[Bridge::SYSTEM_ERRORS] !== []) {
                         $this->invalidateFile($file);
-                    } elseif (! $configuration->isDryRun()) {
+                    } elseif (! $configuration->isDryRun() || $processFileResult->isFileChanged()) {
```

Comment:
I think you can just remove the condition, making it an `else`.
only unchanged files are added as cacheable files, see https://github.com/rectorphp/rector-src/blob/main/src/Application/FileProcessor/PhpFileProcessor.php#L97-L100
and `cacheFileWithDependencies` will do nothing if current file is not a cacheable one, see https://github.com/rectorphp/rector-src/blob/6322d7c7c5bf262524b254cb329644cd051f94b2/packages/Caching/Detector/ChangedFilesDetector.php#L47-L49